### PR TITLE
Make pupload work with #tree.

### PR DESCRIPTION
**Pull request by jordandukart**

### CODE REVIEW

File: islandora_plupload.module [hunk 1]
```diff
@@ -99,16 +91,18 @@ function islandora_plupload_alter_managed_files(&$form) {
  */
 function islandora_plupload_element_validate($element, &$form_state) {
   module_load_include('inc', 'islandora', 'includes/mime_detect');
-  if (isset($element['#type']) && $element['#type'] == 'plupload' && isset($form_state['values'][$element['#name']][0]) && is_array($form_state['values'][$element['#name']][0])) {
-    $file = plupload_file_uri_to_object($form_state['values'][$element['#name']][0]["tmppath"]);
-    $file->filename = $form_state['values'][$element['#name']][0]['name'];
+  $element_value =& drupal_array_get_nested_value($form_state['values'], $element['#parents']);
```
Comment (nigelgbanks) [inline islandora_plupload.module:94]: What if #tree is FALSE? $element['#value'] should contain what we seek.

File: islandora_plupload.module [hunk 2]
```diff
@@ -99,16 +91,18 @@ function islandora_plupload_alter_managed_files(&$form) {
...
+  $first_element_value = reset($element_value);
+  if (isset($element['#type']) && $element['#type'] == 'plupload' && $first_element_value && is_array($first_element_value)) {
```
Comment (nigelgbanks) [inline islandora_plupload.module]: Do we need to test the #type? Shouldn't this element validate callback only ever be attached to a plupload element?

File: islandora_plupload.module (hunk 1 above)
Comment (jordandukart) [inline islandora_plupload.module:94]: Tested this with `#tree => FALSE` and `#tree => TRUE` in the audio upload form and didn't have an issue.

Comment (adam-vessey) [inline islandora_plupload.module:94]: Yeah, `#parents` should be good... Thing that I always get confused with is the differences between `#parents` and `#array_parents`, but it [looks like `#parents` is the way to go](https://drupal.org/comment/2292322#comment-2292322).

**Pull request merged**